app/uk/gov/hmrc/gform/sharedmodel/formtemplate/EmailParameter.scala

```diff
@@ -53,3 +54,20 @@ object EmailParametersRecalculated {
   implicit val format: OFormat[EmailParametersRecalculated] = Json.format[EmailParametersRecalculated]
 
 }
+
+case class EmailCodeParameter(emailTemplateVariable: String, value: LocalisedString)
+object EmailCodeParameter {
+  implicit val format: OFormat[EmailCodeParameter] = {
+    val mongoFormat = Json.format[EmailCodeParameter]
+
+    val uploadTemplateReads: Reads[EmailCodeParameter] =
+      for {
+        emailTemplateVariable <- (JsPath \ "emailTemplateVariable").read[String]
+        value                 <- (JsPath \ "value").read[LocalisedString]
+      } yield EmailCodeParameter(emailTemplateVariable, value)
+
+    val reads: Reads[EmailCodeParameter] = uploadTemplateReads | mongoFormat
+
+    OFormat(reads, mongoFormat)
```

Comment:
There is `OFormatWithTemplateReadFallback` for this

Reply:
I simplified